app/controllers/products_controller.rb

```diff
@@ -41,7 +41,7 @@ def create
     @new_method = @type == :usr ? 'git' : params[:new_method]
 
     respond_to do |format|
-      if @product.save
+      if @product.save(context: @product.create_context)
```

Comment:
Instead of adding create_context as an attribute to Product, it makes more sense here that create_context would be just on the form and in params. So if you do this you might use `hidden_field` instead of `f.hidden_field` and then in params you could access it: `params[:create_context]` i.e. `@product.save(context: params[:create_context]`.